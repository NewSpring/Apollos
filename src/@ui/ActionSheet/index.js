export { ActionSheetProvider, connectActionSheet } from '@expo/react-native-action-sheet';
