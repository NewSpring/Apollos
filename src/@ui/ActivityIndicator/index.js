import { ActivityIndicator as NativeActivityIndicator, StyleSheet } from 'react-native';
import styled from '@ui/styled';

const ActivityIndicator = styled({
  flex: 1,
  ...StyleSheet.absoluteFillObject,
  alignSelf: 'center',
}, 'ActivityIndicator')(NativeActivityIndicator);

export {
  ActivityIndicator as default,
  NativeActivityIndicator as InlineActivityIndicator,
};
