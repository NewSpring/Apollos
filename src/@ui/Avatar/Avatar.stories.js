import React from 'react';

import { storiesOf } from '@storybook/react-native';
import FlexedView from '@ui/FlexedView';

import Avatar from './';

const source = { url: 'https://dg0ddngxdz549.cloudfront.net/images/cached/images/remote/http_s3.amazonaws.com/ns.images/all/member_images/members.nophoto_1000_1000_90_c1.jpg' };

storiesOf('Avatar', module)
  .add('default', () => (
    <FlexedView style={{ flexWrap: 'wrap', flexDirection: 'row' }}>
      <Avatar
        source={source}
        size="small"
      />
      <Avatar
        source={source}
        size="medium"
      />
      <Avatar
        source={source}
        size="large"
      />
    </FlexedView>
  ));

