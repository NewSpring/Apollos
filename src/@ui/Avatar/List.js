import { View } from 'react-native';
import styled from '@ui/styled';

const AvatarList = styled({
  flexDirection: 'row',
  flexWrap: 'wrap',
  maxWidth: '100%',
}, 'Avatar.List')(View);

export default AvatarList;
