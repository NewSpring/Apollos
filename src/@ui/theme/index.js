export { default as ThemeProvider } from './ThemeProvider';
export { default as withTheme } from './withTheme';
export { ThemeMixin, withThemeMixin } from './mixins';
