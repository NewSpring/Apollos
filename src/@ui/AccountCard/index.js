import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes, defaultProps } from 'recompose';
import { Platform } from 'react-native';
import { startCase, toLower } from 'lodash';

import { withIsLoading } from '@ui/isLoading';
import styled from '@ui/styled';
import Card, { CardContent } from '@ui/Card';
import SideBySideView from '@ui/SideBySideView';
import FlexedView from '@ui/FlexedView';
import { H5 } from '@ui/typography';
import CategoryLabel from '@ui/CategoryLabel';
import Icon from '@ui/Icon';
import last4 from '@utils/last4';
import { withTheme } from '@ui/theme';

const enhance = compose(
  setPropTypes({
    title: PropTypes.string.isRequired,
    isLoading: PropTypes.bool,
    accountNumber: PropTypes.number,
    accountType: PropTypes.number,
    iconSize: PropTypes.number,
  }),
  defaultProps({
    isLoading: false,
    accountNumber: '',
    accountType: 'creditCard',
    iconSize: undefined,
  }),
  withIsLoading,
  withTheme(({ theme, ...otherProps }) => ({
    iconSize: otherProps.iconSize || theme.helpers.rem(1),
  })),
  pure,
);

const HorizontalLayout = styled({
  alignItems: 'center',
  minHeight: 110, // kind of the best middle ground for various title lengths.
})(SideBySideView);

const LeftColumn = styled({
  flex: 1.66,
})(CardContent);

const RightColumn = styled(({ theme }) => ({
  paddingRight: theme.sizing.baseUnit,
  alignItems: 'flex-end',
  justifyContent: 'center',
  ...Platform.select({
    web: {
      position: 'relative',
    },
  }),
}))(FlexedView);

const AccountCard = enhance(({
  title,
  isLoading,
  accountNumber,
  accountType,
  iconSize,
  ...otherProps
}) => (
  <Card isLoading={isLoading} {...otherProps}>
    <HorizontalLayout>
      <LeftColumn>
        <H5>{startCase(toLower(title))}</H5>
        <CategoryLabel
          label={last4(accountNumber)}
          icon={accountType === 'bankAccount' ? 'bank' : 'credit'}
          isLoading={isLoading}
        />
      </LeftColumn>

      <RightColumn>
        <Icon name="ArrowNext" size={iconSize} />
      </RightColumn>
    </HorizontalLayout>
  </Card>
));

export default AccountCard;
