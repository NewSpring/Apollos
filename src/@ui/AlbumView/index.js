import React from 'react';
import PropTypes from 'prop-types';
import { compose, pure, setPropTypes } from 'recompose';
import { View, Platform, StyleSheet } from 'react-native';
import PaddedView from '@ui/PaddedView';
import ProgressiveImage from '@ui/ProgressiveImage';
import { H5, H7 } from '@ui/typography';
import { withThemeMixin } from '@ui/theme';
import styled from '@ui/styled';

const Container = styled(({ theme }) => ({
  backgroundColor: theme.colors.background.default,
}), 'AlbumView.Container')(View);

const InfoWrapper = styled({
  flexDirection: 'row', alignItems: 'stretch',
}, 'AlbumView.InfoWrapper')(PaddedView);

const AlbumArt = styled({
  position: 'relative',
  width: '33%',
  aspectRatio: 1,
  ...Platform.select({
    web: {
      height: 0,
      paddingTop: '100%',
    },
  }),
}, 'AlbumView.AlbumArt')(ProgressiveImage);

const BlurredImage = styled(({ theme }) => ({
  ...StyleSheet.absoluteFillObject,
  opacity: theme.alpha.low,
}), 'AlbumView.BlurredImage')(ProgressiveImage);

const TitleWrapper = styled({
  width: '66%',
  justifyContent: 'space-around',
}, 'AlbumView.TitleWrapper')(PaddedView);

const Title = styled({
  backgroundColor: 'transparent',
  flexWrap: 'wrap',
}, 'AlbumView.Title')(H5);

const Artist = styled({
  backgroundColor: 'transparent',
  flexWrap: 'wrap',
}, 'AlbumView.Artist')(H7);

const enhance = compose(
  withThemeMixin({ type: 'dark' }),
  pure,
  setPropTypes({
    title: PropTypes.string,
    artist: PropTypes.string,
    blurredImage: ProgressiveImage.propTypes.source,
    albumImage: ProgressiveImage.propTypes.source,
    isLoading: PropTypes.bool,
  }),
);

const AlbumView = enhance(({
  title, artist, blurredImage, albumImage, isLoading = false,
}) => (
  <Container>
    {(blurredImage) ? <BlurredImage source={blurredImage} /> : null}
    <InfoWrapper>
      <AlbumArt isLoading={isLoading} source={albumImage} />
      <TitleWrapper>
        <Title>{title}</Title>
        <Artist>{artist}</Artist>
      </TitleWrapper>
    </InfoWrapper>
  </Container>
));

export default AlbumView;
