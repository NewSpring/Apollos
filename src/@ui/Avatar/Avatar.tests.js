import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from '@ui/theme';

import Avatar from './';

const source = { url: 'https://dg0ddngxdz549.cloudfront.net/images/cached/images/remote/http_s3.amazonaws.com/ns.images/all/member_images/members.nophoto_1000_1000_90_c1.jpg' };

describe('The Avatar component', () => {
  it('should render small', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Avatar
          source={source}
          size="small"
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render medium', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Avatar
          source={source}
          size="medium"
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render large', () => {
    const tree = renderer.create(
      <ThemeProvider>
        <Avatar
          source={source}
          size="large"
        />
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
