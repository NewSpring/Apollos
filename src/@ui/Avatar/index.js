import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { get } from 'lodash';
import { compose, pure, setPropTypes } from 'recompose';

import ConnectedImage from '@ui/ConnectedImage';
import styled from '@ui/styled';
import { withTheme } from '@ui/theme';
import ActivityIndicator from '@ui/ActivityIndicator';

export { default as AvatarList } from './List';

const enhance = compose(
  pure,
  setPropTypes({
    size: PropTypes.oneOf(['small', 'medium', 'large']),
    containerStyle: PropTypes.any, // eslint-disable-line
    ...ConnectedImage.propTypes,
  }),
  withTheme(({ theme, size }) => ({
    themeSize: get(theme.sizing.avatar, size, theme.sizing.avatar.small),
  })),
);

const Container = styled(
  ({ theme, themeSize }) => ({
    width: themeSize,
    height: themeSize,
    backgroundColor: theme.colors.background.inactive,
    borderRadius: themeSize / 2,
    marginRight: themeSize / 4,
    marginBottom: themeSize / 4,
    overflow: 'hidden',
  }),
  'Avatar',
)(View);

const LoadingIcon = compose(
  withTheme(({ theme: { colors } = {} }) => ({ color: colors.white })),
  styled({
    zIndex: 1,
  }),
)(ActivityIndicator);

const Image = styled(({ themeSize }) => ({
  width: '100%',
  height: '100%',
  position: 'absolute',
  top: 0,
  left: 0,
  right: 0,
  bottom: 0,
  borderRadius: themeSize / 2,
}))(ConnectedImage);

const Avatar = enhance(({
  themeSize, containerStyle, source, isLoading, ...imageProps
}) => (
  <Container style={containerStyle} themeSize={themeSize}>
    {isLoading ? <LoadingIcon /> : null}
    {source ? (
      <Image source={source} {...imageProps} themeSize={themeSize} isLoading={isLoading} />
    ) : null}
  </Container>
));

export default Avatar;
