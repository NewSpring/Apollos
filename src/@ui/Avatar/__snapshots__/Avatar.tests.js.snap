// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Avatar component should render large 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#dddddd",
      "borderRadius": 80,
      "height": 160,
      "marginBottom": 40,
      "marginRight": 40,
      "overflow": "hidden",
      "width": 160,
    }
  }
  themeSize={160}
>
  <View
    style={
      Object {
        "aspectRatio": 1,
        "backgroundColor": "#dddddd",
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`The Avatar component should render medium 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#dddddd",
      "borderRadius": 40,
      "height": 80,
      "marginBottom": 20,
      "marginRight": 20,
      "overflow": "hidden",
      "width": 80,
    }
  }
  themeSize={80}
>
  <View
    style={
      Object {
        "aspectRatio": 1,
        "backgroundColor": "#dddddd",
        "width": "100%",
      }
    }
  />
</View>
`;

exports[`The Avatar component should render small 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#dddddd",
      "borderRadius": 20,
      "height": 40,
      "marginBottom": 10,
      "marginRight": 10,
      "overflow": "hidden",
      "width": 40,
    }
  }
  themeSize={40}
>
  <View
    style={
      Object {
        "aspectRatio": 1,
        "backgroundColor": "#dddddd",
        "width": "100%",
      }
    }
  />
</View>
`;
